pkg/kubelet/events/resize.go

```diff
@@ -26,38 +26,150 @@ import (
+	if err != nil {
+		klog.ErrorS(err, "Failed to serialize resource summary", "pod", format.Pod(pod))
+		return "Pod resize completed"
 	}
-	type podResourceSummary struct {
-		//TODO: resources v1.ResourceRequirements, add pod-level resources here once resizing pod-level resources is supported
-		InitContainers []containerAllocation `json:"initContainers,omitempty"`
-		Containers     []containerAllocation `json:"containers,omitempty"`
+	podResizeCompletedMsg := fmt.Sprintf("Pod resize completed: %s", string(podResizeMsgDetailsJSON))
+	return podResizeCompletedMsg
+}
+
+// PodResizeInProgressMsg generates the pod resize in progress event message.
+func PodResizeInProgressMsg(allocatedPod *v1.Pod, observedGeneration int64) string {
+	type podResizeDiff struct {
+		ObservedGeneration int64               `json:"observedGeneration"`
+		ActualResources    *podResourceSummary `json:"actual,omitempty"`
+		AllocatedResources *podResourceSummary `json:"allocated,omitempty"`
 	}
 
-	podResizeSource := &podResourceSummary{}
+	allocated, actual := makeResourceSummary(allocatedPod)
 
-	for container, containerType := range podutil.ContainerIter(&allocatedPod.Spec, podutil.InitContainers|podutil.Containers) {
+	diff := &podResizeDiff{
+		ObservedGeneration: observedGeneration,
+		AllocatedResources: allocated,
+		ActualResources:    actual,
+	}
+
+	podResizeMsgDetailsJSON, err := json.Marshal(diff)
+	if err != nil {
+		klog.ErrorS(err, "Failed to serialize resource summary", "pod", format.Pod(allocatedPod))
+		return "Pod resize in progress"
+	}
+	podResizeInProgressMsg := fmt.Sprintf("Pod resize in progress: %s", string(podResizeMsgDetailsJSON))
+	return podResizeInProgressMsg
+}
+
+// PodResizeInProgressErrorMsg generates the pod resize in progress error event message.
+func PodResizeInProgressErrorMsg(allocatedPod *v1.Pod, observedGeneration int64, errorMsg string) string {
+	type podResizeDiff struct {
+		ObservedGeneration int64               `json:"observedGeneration"`
+		ActualResources    *podResourceSummary `json:"actual,omitempty"`
+		AllocatedResources *podResourceSummary `json:"allocated,omitempty"`
+		Error              string              `json:"error"`
+	}
+
+	allocated, actual := makeResourceSummary(allocatedPod)
+	diff := &podResizeDiff{
+		ObservedGeneration: observedGeneration,
+		AllocatedResources: allocated,
+		ActualResources:    actual,
+		Error:              errorMsg,
+	}
+
+	podResizeMsgDetailsJSON, err := json.Marshal(diff)
+	if err != nil {
+		klog.ErrorS(err, "Failed to serialize resource summary", "pod", format.Pod(allocatedPod))
+		return "Pod resize in progress reported an error: " + errorMsg
+	}
+	podResizeInProgressMsg := fmt.Sprintf("Pod resize in progress reported an error: %s", string(podResizeMsgDetailsJSON))
+	return podResizeInProgressMsg
+}
+
+// PodResizePendingMsg generates the pod resize pending event message.
+func PodResizePendingMsg(pod, allocatedPod *v1.Pod, reason string, observedGeneration int64) string {
```

Comment:
Do you think this event should include the message component of the pending condition? Or did you decide not to do that since we only fire the event once? It seems like it would at least be useful for infeasible, in which case the message shouldn't change.